src/manager.cc

```diff
@@ -224,7 +224,13 @@ bool fastcat::Manager::Process(double external_time)
 {
   std::lock_guard<std::mutex> lock(parameter_mutex_);
   for (auto it = jsd_map_.begin(); it != jsd_map_.end(); ++it) {
-    jsd_read(it->second, 1e6 / target_loop_rate_hz_);
+    auto ifname = it->first;
+    auto jsd = it->second;
+    jsd_read(jsd, 1e6 / target_loop_rate_hz_);
+    if (jsd->wkc != jsd->expected_wkc && !IsFaulted()) {
+      ERROR("Bad working counter experienced on jsd bus %s", ifname.c_str());
+      ExecuteAllDeviceFaults();
+    }
```

Comment:
Nice! simple and effective solution to WKC faults. I'm curious to hear how this pans out with testing!